File: src/main/java/net/b0n541/pmcts/game/skat/PlayerPosition.java
package net.b0n541.pmcts.game.skat;

public enum PlayerPosition {
    FOREHAND,
    MIDDLEHAND,
    REARHAND;

    public PlayerPosition getNextPlayer() {
        final PlayerPosition result;
        switch (this) {
            case FOREHAND:
                result = MIDDLEHAND;
                break;
            case MIDDLEHAND:
                result = REARHAND;
                break;
            case REARHAND:
                result = FOREHAND;
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + this);
        }
        return result;
    }
}

File: src/main/kotlin/net/b0n541/pmcts/game/skat/PlayerPosition.kt
package net.b0n541.pmcts.game.skat

enum class PlayerPosition {
    FOREHAND, MIDDLEHAND, REARHAND;

    val nextPlayer: PlayerPosition
        get() {
            return when (this) {
                FOREHAND -> MIDDLEHAND
                MIDDLEHAND -> REARHAND
                REARHAND -> FOREHAND
            }
        }
}
